app/src/main/java/emu/skyline/input/InputHandler.kt

```diff
@@ -60,6 +80,28 @@ class InputHandler(private val inputManager : InputManager, private val preferen
         external fun setTouchState(points : IntArray)
     }
 
+    data class MotionSensorInput(
+        var timestamp : u64 = 0uL,
+        var deltaTimestamp : u64 = 0uL,
+        @param:ByteBufferSerializable.ByteBufferSerializableArray(3) var gyroscope : FloatArray = FloatArray(3),
+        @param:ByteBufferSerializable.ByteBufferSerializableArray(3) var accelerometer : FloatArray = FloatArray(3),
+    ) : ByteBufferSerializable
+
+    /**
+     * The latest state of the motion sensor
+     */
+    private val motionSensor = MotionSensorInput()
+
+    /**
+     * Buffer for passing motion data to c++
+     */
+    private val motionDataBufferLeft = ByteBuffer.allocateDirect(0x28).order(ByteOrder.LITTLE_ENDIAN)
+    private val motionDataBufferRight = ByteBuffer.allocateDirect(0x28).order(ByteOrder.LITTLE_ENDIAN)
+    private val motionDataBufferHandheld = ByteBuffer.allocateDirect(0x28).order(ByteOrder.LITTLE_ENDIAN)
```

Comment:
See the other comment. Also, create a `MotionSensorInputDataSize` constant with value `0x28` and use that here to avoid the magic number.